Supporting Pimoroni Pico Plus2

soburi opened the pull request:
Add support Pimoroni Pico Plus2 https://shop.pimoroni.com/products/pimoroni-pico-plus-2?variant=42092668289107
and RP2350B.

~~Note: #84974 need to be addressed first.~~

📄 Docs preview: https://builds.zephyrproject.io/zephyr/pr/77859/docs/boards/pimoroni/pico_plus2/doc/index.html

soburi committed:
boards: pimoroni: pico_plus2: Add Pimoroni Pico Plus2 board

Pinoroni Pico Plus2 is an RP2350B based board that has
more rich I/O, RAM, and Flash than the original Raspberry Pi Pico2.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

nordicjm commented on boards/pimoroni/pico_plus2/doc/index.rst:
```diff
@@ -0,0 +1,303 @@
+Supported Features
+==================
+
+The rpi_pico board configuration supports the following
```
board name as used by west with \`\` around it and board target, not configuration

soburi replied:
Completely rewritten, please check again.

nordicjm commented on boards/pimoroni/pico_plus2/pico_plus2-pinctrl.dtsi:
```diff
@@ -0,0 +1,52 @@
+		};
+	};
+
+	clocks_default: clocks_default {
```
empty can go

soburi replied:
Same as above.

soburi committed:
soc: raspberrypi: rp2350: Add initial support for the RP2350B

RP2350B is a rich IO variant in the RP2350 series.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
drivers: gpio: rpi_pico: Add support for RP2350B

As the RP2350B has more than 32 GPIO pins,
we changed the configuration so that it is split into two ports.

To do this, we created a `raspberrypi,pico-gpio-port` node and
moved the previous `raspberrypi,pico-gpio-port` functions to it.

This became a child node of `raspberrypi,pico-gpio-port`, and
`raspberrypi,pico-gpio-port` will remain a gpio mapper.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

ajf58 commented on tests/drivers/gpio/gpio_basic_api/boards/rpi_pico.overlay:
```diff
@@ -0,0 +1,13 @@
+	resources {
+		compatible = "test-gpio-basic-api";
+		out-gpios = <&pico_header 14 GPIO_ACTIVE_HIGH>;
+		in-gpios = <&pico_header 15 GPIO_ACTIVE_HIGH>;
```
This will clash with the external pulldown resistor tests, which is why in https://github.com/ajf58/zephyr/commit/0b86b90cfd46aaed878dde538ec7c4de3abfd481 I use a different pair of pins. I think you should do the same, otherwise everyone needs to either run a subset of tests at a time, and swap wires round in between, or similar.

soburi replied:
Thank you for your pointing out.
I have addressed this in #83427.

https://github.com/zephyrproject-rtos/zephyr/pull/83427/files#diff-6263b0be3639b44643145774c633549394a54c6234f683cb34e415a71fedc950R10-R11

soburi commented on boards/pimoroni/index.rst:
Fixed.

soburi commented on boards/pimoroni/pico_plus2/doc/index.rst:
```diff
@@ -0,0 +1,111 @@
+     :alt: Pimoroni Pico Plus 2
+
+     Pimoroni Pico Plus 2 (Image courtesy of Pimoroni)
+
```
Fixed.

nordicjm commented on boards/pimoroni/pico_plus2/pico_plus2-pinctrl.dtsi:
```diff
@@ -0,0 +1,49 @@
+		group1 {
+			pinmux = <UART0_TX_P0>;
+		};
+		group2 {
```
spacing

soburi replied:
Fixed it.

nordicjm commented on boards/pimoroni/pico_plus2/pico_plus2.dtsi:
```diff
@@ -0,0 +1,175 @@
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
```
spacing, apply throughout

soburi replied:
Fixed it.

nordicjm commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -238,6 +238,11 @@ GPIO
 * Renamed the device tree property ``off_val`` to ``off-val``.
 * Renamed the device tree property ``on_val`` to ``on-val``.
 * Renamed the ``compatible`` from ``ti,ads114s0x-gpio`` to :dtcompatible:`ti,ads1x4s0x-gpio`.
+* To support the RP2350B, which has many pins, the RaspberryPi-GPIO configuration has
```
4.1 feature freeze is past

soburi replied:
Moved into `doc/releases/migration-guide-4.1.rst`.

petejohanson commented:
@soburi Did you intend to include the Renesas related commits in your recent push?

Looking forward to this change, since I'm planning to submit a board definition for the Adafruit Metro RP2350 which uses RP2350B. I've got it working mostly already, minus the GPIO changes, but would rather piggyback on this core work. Thanks!

soburi commented:
@petejohanson 

> @soburi Did you intend to include the Renesas related commits in your recent push?

Oh! It's my bad. 
I was working on both in parallel so I made a mistake. Thank you for pointing that out.
I fix it ASAP.

nordicjm commented on boards/pimoroni/pico_plus2/pico_plus2_rp2350b_m33_defconfig:
```diff
@@ -0,0 +1,14 @@
+CONFIG_GPIO=y
+CONFIG_RESET=y
+CONFIG_SERIAL=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=150000000
```
as per other PR, should be in soc Kconfig, can be done in a later PR

kartben commented:
@soburi it looks like you may want to update the Raspberry Pi Pico Platforms section in MAINTAINERS.yml file

soburi commented:
@kartben 

> @soburi it looks like you may want to update the Raspberry Pi Pico Platforms section in MAINTAINERS.yml file

It seems that many other boards need to be addressed besides this one, so I would like to address them all together in a separate PR.
